src/app/pages/wallets/my-wallets/pages-wallets-my-wallets.component.ts

```diff
@@ -26,15 +30,65 @@ export class PagesWalletsMyWalletsComponent implements OnInit {
           data,
           isLoading: false,
           hasError: false,
+          nowEditWallet: 0,
         };
       },
       error: () => {
         this.myWalletsData = {
           data: null,
           isLoading: false,
           hasError: true,
+          nowEditWallet: 0,
         };
       },
     });
   }
+
+  public addNewWallet(userInputData_: string) {
+    this.myWalletsService.addNewWallet(userInputData_).subscribe( (wallet_: MyWallet) => {
+        this.myWalletsData.data?.push(wallet_);
+    });
+  }
+
+  public updateWallet(userInputData_: string) {
+    this.myWalletsService.updateNewWallet(this.myWalletsData.nowEditWallet, userInputData_).subscribe( (wallet_: MyWallet) => {
+        const walletIndex = this.myWalletsData.data?.findIndex(w => w.id === this.myWalletsData.nowEditWallet);
+        if(walletIndex != undefined && walletIndex > -1) {
+            this.myWalletsData.data?.splice(walletIndex, 1, wallet_);
+        }
+    });
+  }
+
+  public openWalletModalCallback(id_: number): void {
+    this.myWalletsData.nowEditWallet = id_;
+    this.openWalletModal(id_);
+  }
+
+  public openAddNewWalletModal(): void {
+    this.myWalletsData.nowEditWallet = 0;
+    this.openWalletModal();
+  }
+
+  public openWalletModal(id_: number = 0): void {
+    const walletNameInput = id_ == 0 ? "" : this.myWalletsData.data?.find(w => w.id === id_)?.name;
+    const dialogRef = this.dialog.open(WalletFormModalComponent, {
+        width: '350px',
+        height: '300px',
+        data: { title: 'Update Wallet', isNewWallet: id_ === 0, inputData: walletNameInput, }
```

Comment:
Here basically you should only pass wallet to the modal. Title you don't need -> if wallet you will pass existing wallet to modal that means you are updating it -> that means you can display there title "Updating wallet". If you won't pass wallet to modal that means you are creating new modal -> that means title can be "Creating wallet".
`isNewAllet` you also don't need -> if you won't pass wallet that means you are creating new wallet.
`inputData` you also don't need -> if you will pass wallet object you can read wallet's name in the modal. Also we we will want to add for example later new input for wallet description we won't have to touch this line as we will be already passing whole wallet.